### PR TITLE
Frontend/fix/#780/UI changes rate card

### PR DESCRIPTION
**Pull request by Akash2172**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Changes made in UI of Rate Card at Customer Side

https://drive.google.com/file/d/1DR-E0GrdFEjjLJscGdsf-RLXi1FGZYIF/view?usp=sharing
https://drive.google.com/file/d/1KjkXAnYzC45Yuq3wR9eDc0ctlSq2fz8A/view?usp=sharing

### CODE REVIEW

File: Frontend/android-native/app/src/user/res/drawable/ny_ic_driver_addition_table.png
Comment (witcher-shailesh) [inline Frontend/android-native/app/src/user/res/drawable/ny_ic_driver_addition_table.png:1]: This should be a view, not an image. These values can differ in future so please try to create this component

File: Frontend/ui-customer/src/Screens/Types.purs [hunk 1]
```diff
@@ -570,14 +570,21 @@ type Contact = {
...
 type RateCard =
   {
     baseFare :: Int,
     extraFare :: Int,
     pickUpCharges :: Int,
     additionalFare :: Int,
     nightShiftMultiplier :: Number,
-    nightCharges :: Boolean
+    nightCharges :: Boolean,
+    toggleRateCardView :: RateCardType,
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/Types.purs:586]: what does toggleRateCardView does?

**Review by witcher-shailesh — CHANGES_REQUESTED** [review]
squash the commits

File: Frontend/ui-customer/src/Screens/Types.purs [hunk 2]
```diff
@@ -570,14 +570,21 @@ type Contact = {
      phoneNo :: String
 }
 
+data RateCardType = ViewStart | ViewDriverAddition | ViewFareUpdate
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/Types.purs:573]: Change the following
 ViewStart -> Default
 ViewDriverAddition -> DriverAddition
 ViewFareUpdate -> FareUpdate

File: Frontend/ui-customer/src/Screens/Types.purs [hunk 3]
```diff
@@ -570,14 +570,21 @@ type Contact = {
...
   {
     baseFare :: Int,
     extraFare :: Int,
     pickUpCharges :: Int,
     additionalFare :: Int,
     nightShiftMultiplier :: Number,
-    nightCharges :: Boolean
+    nightCharges :: Boolean,
+    toggleRateCardView :: RateCardType,
+    isCurrentStart :: Boolean
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/Types.purs:587]: Change the `isCurrentStart` to understandable variable

File: Frontend/ui-customer/src/Components/RateCard/View.purs [hunk 4]
```diff
@@ -83,147 +92,365 @@ view push config =
...
+        , text if config.toggleRateCardView == ViewStart then (getString GOT_IT) else (getString GO_BACK_)
+        , textSize FontSize.a_18
+        , lineHeight "20"
+        , padding (Padding 0 8 0 25)
+        , onClick push if config.toggleRateCardView == ViewStart then const Close else const GoToViewStart
+        ]
+      ]
+    ]      
+  ]
+viewStart :: forall w. (Action -> Effect Unit) -> Config -> PrestoDOM (Effect Unit) w
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Components/RateCard/View.purs:123]: change the `viewStart` to `view_name <> View`  like `defaultRateCardView`. Check all the places and  change to understandable names.

File: Frontend/ui-customer/src/Components/RateCard/View.purs [hunk 5]
```diff
@@ -48,30 +52,35 @@ view push config =
...
            ][ textView
               [ width WRAP_CONTENT
               , height WRAP_CONTENT
               , textSize FontSize.a_24
+              , lineHeight "28"
               , color if config.nightCharges then Color.white900 else Color.black800
               , text (getString RATE_CARD)
               , fontStyle $ FontStyle.bold LanguageStyle
-              , margin (Margin 0 5 0 5)
+              , margin (Margin 0 4 0 4)
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Components/RateCard/View.purs:75]: use `MarginVertical`

File: Frontend/ui-customer/src/Components/RateCard/View.purs [hunk 6]
```diff
@@ -48,30 +52,35 @@ view push config =
...
               ]
             , textView
               [ width WRAP_CONTENT
               , height WRAP_CONTENT
               , textSize FontSize.a_14
+              , lineHeight "15"
+              , fontStyle $ FontStyle.medium LanguageStyle
+              , padding (Padding 0 0 0 10)
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Components/RateCard/View.purs:83]: use `PaddingBottom` check all the places for the same

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/Types.purs [hunk 7]
```diff
@@ -570,14 +570,21 @@ type Contact = {
      phoneNo :: String
 }
 
+data RateCardType = ViewStart | ViewDriverAddition | ViewFareUpdate
+derive instance genericRateCardType :: Generic RateCardType _
+instance showRateCardType :: Show RateCardType where show = genericShow
```
[the remark this answers is not in the document]
Comment (Akash2172) [reply Frontend/ui-customer/src/Screens/Types.purs:575]: No

File: Frontend/ui-customer/src/Screens/Types.purs (hunk 1 above)
Comment (Akash2172) [reply Frontend/ui-customer/src/Screens/Types.purs:586]: I changed the name to currentRateCardType after discussing it with Vignesh bhaiya

File: Frontend/ui-customer/src/Screens/Types.purs (hunk 7 above)
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/ui-customer/src/Screens/Types.purs:575]: if it not required remove the instance.

File: Frontend/ui-customer/src/Components/RateCard/View.purs [hunk 8]
```diff
@@ -83,147 +92,361 @@ view push config =
...
+      , lineHeight "16"
+      , fontStyle $ FontStyle.regular LanguageStyle
+      , color Color.black650
+      , text (getString DRIVER_MAY_NOT_CHARGE_THIS_ADDITIONAL_FARE)
+      , padding (Padding 20 0 20 24)
+      , margin $ MarginTop 12
+      ]
+    ]
+
+FareUpdateView :: forall w. (Action -> Effect Unit) -> Config -> PrestoDOM (Effect Unit) w
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Components/RateCard/View.purs:397]: functions name should not start with capital letters

File: Frontend/ui-customer/src/Components/RateCard/View.purs [hunk 9]
```diff
@@ -83,147 +92,361 @@ view push config =
...
-           [ width WRAP_CONTENT
-           , height WRAP_CONTENT
-           , textSize FontSize.a_16
-           , color Color.black800
-           , text config.baseFare
-           , gravity RIGHT
-           , weight 1.0
-           ]
-         ]
+        ][PrestoAnim.animationSet [ if config.currentRateCardType == DefaultRateCard then (translateInXBackwardAnim if config.onFirstPage == true then true else false) else (translateInXForwardAnim true) ] $
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Components/RateCard/View.purs:99]: `config.onFirstPage` this you can directly assign this value no need of this `if config.onFirstPage == true then true else false`

File: Frontend/ui-customer/src/Components/RateCard/View.purs [hunk 10]
```diff
@@ -83,147 +92,361 @@ view push config =
...
+        [ width MATCH_PARENT
+        , height WRAP_CONTENT
+        , color Color.blue800
+        , gravity CENTER
+        , fontStyle $ FontStyle.semiBold LanguageStyle
+        , text if config.currentRateCardType == DefaultRateCard then (getString GOT_IT) else (getString GO_BACK_)
+        , textSize FontSize.a_18
+        , lineHeight "20"
+        , padding (Padding 0 8 0 25)
+        , onClick push if config.currentRateCardType == DefaultRateCard then const Close else const GoToViewStart
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Components/RateCard/View.purs:118]: `GoToViewStart` change this also.

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**